src/librustdoc/visit_ast.rs

```diff
@@ -309,10 +313,17 @@ impl<'a, 'tcx> RustdocVisitor<'a, 'tcx> {
                     }
                 }
 
-                om.items.push((item, renamed))
+                om.push_item(self.cx, Item::new(item, renamed, from_glob))
             }
             hir::ItemKind::Mod(ref m) => {
-                om.mods.push(self.visit_mod_contents(item.span, &item.vis, item.hir_id(), m, name));
+                om.push_mod(self.visit_mod_contents(
+                    item.span,
+                    &item.vis,
+                    item.hir_id(),
+                    m,
+                    name,
+                    from_glob,
```

Comment:
Hmm, are you sure `from_glob` should apply recursively like this? What happens if you have
```rust
pub mod outer {
  pub const X: usize = 0;
  pub mod inner {
    pub use super::*;
    pub const X: usize = 1;
    pub const Y: usize = 2;
  }
}
pub use inner::*;
```
Shouldn't `inner::X` take precedence over `outer::X`?

Reply:
This could be set as a test case to prevent regression.